### PR TITLE
[documentation] Routes

### PR DESCRIPTION
**Pull request by misstick**

### CODE REVIEW

File: doc/routes.md [hunk 1]
```diff
@@ -70,19 +70,34 @@ Complex filters (in opposite to boolean filters like previous) can have paramete
...
+ - all other stores are object that manipulate data without storing it.
+ - `client/app/mixins/store_watch_mixin` links `ApplicationComponent` to `Stores:change` events.
 
- Each property related to route is changed after `ActionTypes.ROUTE_CHANGE` into `routerStore` file. After all updates are done, the URL is updated if needed :
 
- ```
- currentURL = @getCurrentURL isServer: false
- if location.hash isnt currentURL
-     _router.navigate currentURL
+### Update application from route
```
Comment (misstick) [inline doc/routes.md]: @m4dz, @aenario : could you give me some feedback aout these explanation, please?

File: doc/routes.md [hunk 2]
```diff
@@ -70,19 +70,44 @@ Complex filters (in opposite to boolean filters like previous) can have paramete
...
+What is happening when typing URL `http://localhost:9125/#mailbox/:mailboxID` into your browser?
+
+- `Router.messageList` is matched and dispatch the action: `ROUTE_CHANGE`,
+- `RouterStore` is updated with new status (ie. `mailboxID`, `accountID`).
+- a `RouterStore:change` event is emitted,
+- a new `ApplicationState` is defined from `Getters` (which get data from `Stores`),
+- if `ApplicationState` has changed, `ApplicationComponent` is updated,
+- otherwise `ApplicationComponent` wait for new `RouterStore:change` event.
+
+Only the route component listen to `Stores:changes` to keep component consistency.
```
Comment (m4dz) [inline doc/routes.md]: route > root :wink:

File: doc/routes.md [hunk 3]
```diff
@@ -70,19 +70,44 @@ Complex filters (in opposite to boolean filters like previous) can have paramete
...
- - `MessageID`: the ID of selected message,
- - `query` : filters into messages list,
- - `tab` : change `tabComponent` (ie. `accountEdit`)
+### How works redirection
 
- Whats happening into `routerStore`:
+At first, lets describe routes files:
+ - `client/app/router`: route mapper,
+ - `client/app/stores/router_store`: save status from `RouterActions`,
+ - all other stores are object that manipulate data without storing it.
```
Comment (aenario) [inline doc/routes.md]: I am not sure I understand the meaning of this sentence (without storing it).

**Comment by nono** [comment]
@misstick should we merge this PR?

**Comment by misstick** [comment]
@nono : no need; I will close it; and re-open one next time.

**Pull request closed without merging**